### ISSUE TITLE
Google test main branch

### ISSUE THREAD

**Issue by fraizy22** [mhx/dwarfs#74]
There is currently an error in the build chain with cmake as the google test repo has no longer a branch called master but now a branch called main. I guess that could some how be quick changed.

fs@vs-qcxhp:~/git/dwarfs/build (main) $ cmake .. -DWITH_TESTS=1
-- Checking for module 'fuse>=2.9.9'
--   No package 'fuse' found
-- Checking for module 'libxxhash'
--   No package 'libxxhash' found
CMake Error at CMakeLists.txt:154 (add_subdirectory):
  The source directory

    /home/fs/git/dwarfs/folly

  does not contain a CMakeLists.txt file.

CMake Error at CMakeLists.txt:155 (add_subdirectory):
  The source directory

    /home/fs/git/dwarfs/fbthrift

  does not contain a CMakeLists.txt file.

CMake Error at CMakeLists.txt:157 (add_subdirectory):
  add_subdirectory given source "zstd/build/cmake" which is not an existing
  directory.

-- Configuring done
-- Generating done
-- Build files have been written to: /home/fs/git/dwarfs/build/googletest-download
[ 11%] Performing download step (git clone) for 'googletest'
Klone nach 'googletest-src' ...
fatal: Ungültige Referenz: master
CMake Error at googletest-download/googletest-prefix/tmp/googletest-gitclone.cmake:40 (message):
  Failed to checkout tag: 'master'

gmake[2]: *** [CMakeFiles/googletest.dir/build.make:111: googletest-prefix/src/googletest-stamp/googletest-download] Fehler 1
gmake[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/googletest.dir/all] Fehler 2
gmake: *** [Makefile:103: all] Fehler 2
CMake Error at CMakeLists.txt:183 (message):
  Build step for googletest failed: 2

-- Configuring incomplete, errors occurred!
See also "/home/fs/git/dwarfs/build/CMakeFiles/CMakeOutput.log".

Cheers,

Fred

**Comment by mhx** [comment]
I'm closing this assuming it's fixed. Please feel free to re-open if you still have problems.

**Comment by fraizy22** [comment]
Yes, working fine. Just needed to checkout the submodules as you said. Just one question, how can I exclude directories/files from being included in the image (like mksquashfs supports it) when using mkdwarfs command?

**Comment by mhx** [comment]
> Just one question, how can I exclude directories/files from being included in the image (like mksquashfs supports it) when using the mkdwarfs command?

The only way to do this at the moment is to enable the (experimental) [Python support](https://github.com/mhx/dwarfs#experimental-python-scripting-support) and supply a filter implemented in Python. It's quite possible that this will be replaced with something better in the future, though.
